### ISSUE TITLE
Pickling deque subclasses ignores their `__dict__`.

### ISSUE THREAD

**Issue by DimitrisJim** [RustPython/RustPython#4769]
Caught after updating `test_deque.py`. In RustPython, pickling a deque subclass ignores its `__dict__`:

```python
>>>>> import pickle
>>>>> from collections import deque
>>>>> class Deque(deque): pass
>>>>> d = Deque()
>>>>> d.x = "value"
>>>>> e = pickle.loads(pickle.dumps(pickle.HIGHEST_PROTOCOL))
>>>>> e.__dict__
{}
```

While in Python, `__dict__` is correctly returned as `{'x': 'value'}`.

This doesn't appear to be the case with some other collections like `list` and `dict` so those could probably serve as an inspiration for what might be missing in `deque`.

**Comment by youknowone** [comment]
Probably we are ignoring subclasses `__dict__` for most of our `__reduce__` implementations.

**Comment by DimitrisJim** [comment]
Yeah, I'd bet `collections` should have most of them. Haven't checked the tests but if `deque` showed up now maybe similar ones will appear for the rest of the collections.
